### DISCUSSION TITLE
Update to .Net 6

### DISCUSSION THREAD

**Discussion by jeffcampbellmakesgames** [jeffcampbellmakesgames/Genesis#33] [Announcements]
With .Net Core 3.1 maintenance support ending at in Dec 2022, I'd like to update the Genesis.CLI and plugins to the latest version .Net 6. This would enable a slew of additional features like:

* C#10 and F# 6
* Hot Reload
* LTS support for fixes for three years
* Many performance Improvements from .Net 5 and .Net 6 versions.

I have a branch available `feat/net_6_upgrade` which contains a working version of the CLI/plugins on .Net 6.  So far it does not seem like any breaking changes other than the framework upgrade would be required. If interested, please check it out and test at your leisure. While I don't have an exact date for this upgrade, it's likely to be sooner than later (perhaps a few weeks).

**Comment by 51stpage** [comment]
This is interesting to me.  I'm on Ubuntu 22.04.  [Microsoft](https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#supported-distributions) says that only dotnet6 is supported for me.

Just to make sure I'm testing properly:
1. Install the Genesis and Entitas-Redux packages from Git into my Unity project like I usually do.
2. Download the GenesisCLI.zip from the `feat/net_6_upgrade` branch.
3. Unzip the Genesis-dotnet6 CLI into a different folder.
4. Point the Unity project at the Genesis-dotnet6 folder.
5. Run through the Plugin import process like normal.

I've also tried importing the Genesis `feat/net_6_upgrade` branch with `https://github.com/jeffcampbellmakesgames/genesis.git#feat/net_6_upgrade` but Unity doesn't appear to like that at all.

Current .NET installation configuration
```
❯ dotnet --list-sdks
6.0.109 [/usr/lib/dotnet/dotnet6-6.0.109/sdk]
❯ dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.9 [/usr/lib/dotnet/dotnet6-6.0.109/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.9 [/usr/lib/dotnet/dotnet6-6.0.109/shared/Microsoft.NETCore.App]
```
Do I have that right?
